buku

```diff
@@ -468,8 +468,39 @@ class BukuDb:
 
         self.conn.commit()
 
+    def delete_tag_at_index(self, index, tag_manual):
+        """ Delete tags for bookmark at index
+
+        :param index: int position of record, 0 for all
+        :tag_manual: string of comma-separated tags to delete manually
+        """
+
+        if index == 0:
+            resp = input('Tags will be deleted for ALL bookmarks. Enter \x1b[1my\x1b[21m to confirm: ')
+            if resp != 'y':
+                return
+
+            self.cur.execute('SELECT id, tags FROM bookmarks ORDER BY id ASC')
+        else:
+            self.cur.execute('SELECT id, tags FROM bookmarks WHERE id = ?', (index,))
+
+        resultset = self.cur.fetchall()
+        for row in resultset:
+            tagsc = row[1].strip(DELIMITER).split(DELIMITER)
```

Comment:
Lines from 489 to 497 can be reduced to fewer lines (and less processing) using the following logic:

    tag_to_delete = DELIMITER + tag_to_delete + DELIMITER (if it's not enclosed already, just verify)
    replace tag_to_delete with DELIMITER in tag string of each matching record.